src/prommis/uky/uky_flowsheet.py

```diff
@@ -862,5 +874,782 @@ def display_results(m):
+    m.fs.R_roaster.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_roaster_accounts,
+            "scaled_param": m.fs.R_roaster.duty,
+            "source": 1,
+            "n_equip": 1,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # 3.4 is UKy Roasting - Gas Scrubber
+    R_gas_scrubber_accounts = ["3.4"]
+    m.fs.R_gas_scrubber = UnitModelBlock()
+    m.fs.R_gas_scrubber.gas_rate = Var(
+        initialize=11500, units=units.ft**3 / units.min
+    )
+    m.fs.R_gas_scrubber.gas_rate.fix()
+    m.fs.R_gas_scrubber.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_gas_scrubber_accounts,
+            "scaled_param": m.fs.R_gas_scrubber.gas_rate,
+            "source": 1,
+            "n_equip": 1,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # 3.5 is UKy Roasting - Spray Chamber Quencher (7000-60000 ft**3/min)
+    R_spray_chamber_quencher_accounts = ["3.5"]
+    m.fs.R_spray_chamber_quencher = UnitModelBlock()
+    m.fs.R_spray_chamber_quencher.gas_rate = Var(
+        initialize=11500, units=units.ft**3 / units.min
+    )
+    m.fs.R_spray_chamber_quencher.gas_rate.fix()
+    m.fs.R_spray_chamber_quencher.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_spray_chamber_quencher_accounts,
+            "scaled_param": m.fs.R_spray_chamber_quencher.gas_rate,
+            "source": 1,
+            "n_equip": 3,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # 3.7 is UKy Roasting - Chiller
+    R_chiller_accounts = ["3.7"]
+    m.fs.R_chiller = UnitModelBlock()
+    m.fs.R_chiller.duty = Var(initialize=131, units=units.MBTU / units.hr)
+    m.fs.R_chiller.duty.fix()
+    m.fs.R_chiller.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": R_chiller_accounts,
+            "scaled_param": m.fs.R_chiller.duty,
+            "source": 1,
+            "n_equip": 1,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # TODO: How should this be handled considering there are multiple feeds with varying concentrations?
+    m.fs.feed_input = Var(initialize=500, units=units.ton / units.hr)
+    m.fs.feed_grade = Var(initialize=356.64, units=units.ppm)
```

Comment:
Not sure how to handle this... Is this referring to the feed of the actual plant as opposed to the feed of this flowsheet (what's being fed into the leaching model)? If so, then I assume we can just leave this as is.

Reply:
Shouldn't these be tied to flowsheet variables too? At the moment it seems like the costing is entirely separate from the flowsheet.

Reply:
Yea, I just carried over these values from the costing test file since I wasn't sure what value to use here

Reply:
Yes, `m.fs.feed_input` refers to the total plant feed, which is different than the leaching feed when you have pre-roasting (some mass is lost). For this case I think it would be your flowsheet feed.